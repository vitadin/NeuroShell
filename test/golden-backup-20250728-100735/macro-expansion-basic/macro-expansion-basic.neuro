%% Basic command-level macro expansion test
\set[_echo_command="true"]

%% Test 1: Basic command expansion
\set[cmd="\echo"]
${cmd} Hello World

%% Test 2: Command with variable parameters  
\set[greeting="Hello"]
\set[name="Claude"]
${cmd} ${greeting}, ${name}!

%% Test 3: Different command expansion
\set[getcmd="\get"]
${getcmd}[greeting]

%% Test 4: Command expansion with storage
\set[setcmd="\set"]
${setcmd}[result="Success"]
\get[result]

%% Test 5: Verify expanded commands work correctly
\set[echocmd="\echo"]
${echocmd}[to=output] Testing macro expansion
\get[output]

%% Test 6: Multi-word command expansion
\set[full_echo_cmd="\echo[style=green]"]
${full_echo_cmd} This should be green

%% Test 7: Complex command with options
\set[complex_cmd="\echo[raw=true,to=complex_result]"]
${complex_cmd} Raw\nOutput\tTest
\get[complex_result]