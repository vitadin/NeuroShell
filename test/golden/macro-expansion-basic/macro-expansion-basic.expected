Setting _echo_command = true
%%> \set[cmd="\echo"]
Setting cmd = \echo
%%> \echo Hello World
Hello World
%%> \set[greeting="Hello"]
Setting greeting = Hello
%%> \set[name="Claude"]
Setting name = Claude
%%> \echo Hello, Claude!
Hello, Claude!
%%> \set[getcmd="\get"]
Setting getcmd = \get
%%> \get[greeting]
greeting = Hello
%%> \set[setcmd="\set"]
Setting setcmd = \set
%%> \set[result="Success"]
Setting result = Success
%%> \get[result]
result = Success
%%> \set[echocmd="\echo"]
Setting echocmd = \echo
%%> \echo[to=output] Testing macro expansion
Testing macro expansion
%%> \get[output]
output = Testing macro expansion
%%> \set[full_echo_cmd="\echo[style=green]"]
Setting full_echo_cmd = \echo[style=green]
%%> \echo[style=green] This should be green
This should be green
%%> \set[complex_cmd="\echo[raw=true,to=complex_result]"]
Setting complex_cmd = \echo[raw=true,to=complex_result]
%%> \echo[raw=true,to=complex_result] Raw\nOutput\tTest
Raw\nOutput\tTest
%%> \get[complex_result]
complex_result = Raw\nOutput\tTest